### ISSUE TITLE
script leaks env priv key to terminal w high verbosity

### ISSUE THREAD

**Issue by wminshew** [foundry-rs/foundry#3333]
### Component

Forge

### Have you ensured that all of these are up to date?

- [X] Foundry
- [X] Foundryup

### What version of Foundry are you on?

forge 0.2.0 (3a462eb 2022-09-22T03:37:08.364254Z)

### What command(s) is the bug in?

forge script

### Operating System

macOS (Intel)

### Describe the bug

not sure if this qualifies as a "bug" but I think it's worth discussing. Example cmd/output below for reference

```
λ forge script script/Script.s.sol:MyScript --rpc-url $RPC_URL -vvvv
[⠰] Compiling...
No files changed, compilation skipped
Traces:
  [x] MyScript::run()
    ├─ [0] VM::envUint(PRIVATE_KEY)
    │   └─ ← {{ leaked }}
    ├─ [0] VM::startBroadcast({{ leaked }})
    │   └─ ← ()
    ├─ [x] → new MyFactory@address
    │   ├─ [x] → new MyImpl@address
    │   │   └─ ← x bytes of code
    │   └─ ← x bytes of code
    ├─ [0] VM::stopBroadcast()
    │   └─ ← ()
    └─ ← ()


Script ran successfully.
==========================
Simulated On-chain Traces:

  [x] Create2Deployer::create2()
    ├─ [x] → new MyFactory@address
    │   ├─ [x] → new MyImpl@address
    │   │   └─ ← x bytes of code
    │   └─ ← x bytes of code
    └─ ← address
```

ideally there would be a way to get the trace w/o logging `privKey` to term? but maybe .. maybe these deploy scripts just shouldn't be run w `-vvvv`

(I've always thought having sensitive information in the terminal was less secure than inside a e.g. `.env` but as I'm thinking about it now if someone has access to your terminal screen / output they probably have access to your `.env`? arguably the main downside is the logs might be stored for a long time and searched later, whereas presumably you don't keep your `privKey` stored for a long period of time, only when in use)

**Comment by wminshew** [comment]
not familiar w the internals so don't know if feasible but one possible sol'n is a variant of cheatcode (or flag) that prevents publishing its value or something

**Comment by mattsse** [comment]
thanks for raising this.

fixed in #3336 by redacting the input when we decode cheatcode traces
